### ISSUE TITLE
Expected a boolean but got object

### ISSUE THREAD

**Issue by Pipeliner** [Azure/autorest.typescript#1362]
```
$ npx autorest --typescript a.json
AutoRest code generation utility [cli version: 3.6.1; node: v16.13.1]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
info    | AutoRest core version selected from configuration: ^3.7.0.
info    |    Loading AutoRest core      '/home/vadim/.autorest/@autorestcore@3.8.3/nodemodules/@autorest/core/dist' (3.8.3)
error   | [OperationAbortedException] Error occurred. Exiting.
error   |   Error: [OperationAbortedException] Error occurred. Exiting.
error   | Autorest completed with an error. If you think the error message is unclear, or is a bug, please declare an issues at https://github.com/Azure/autorest/issues with the error message you are seeing.
error   | invalidType | Expected a boolean but got object: {
  title: 'Our title',
  description: 'Our description',
  version: '0.2',
  contact: {}
}
    - file:///var/tmp/pp/scratch/a.json:167:3
```

Minimized Swagger definition:
https://gist.github.com/Pipeliner/0ce99648cf721987262e5579909d7df9

If I remove `info` object from the definition, I get the following error instead:
```
$ npx autorest --typescript a.json
AutoRest code generation utility [cli version: 3.6.1; node: v16.13.1]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
info    | AutoRest core version selected from configuration: ^3.7.0.
info    |    Loading AutoRest core      '/home/vadim/.autorest/@autorestcore@3.8.3/nodemodules/@autorest/core/dist' (3.8.3)
info    |    Loading AutoRest extension '@autorest/typescript' (latest->6.0.0-beta.19)
info    |    Loading AutoRest extension '@autorest/modelerfour' (4.21.4->4.21.4)
error   | [Exception] No input files provided.

Use --help to get help information or see https://aka.ms/autorest/cli for additional documentation
error   |   Error: [Exception] No input files provided.

Use --help to get help information or see https://aka.ms/autorest/cli for additional documentation
error   | Autorest completed with an error. If you think the error message is unclear, or is a bug, please declare an issues at https://github.com/Azure/autorest/issues with the error message you are seeing.

```

**Comment by Pipeliner** [comment]
Same as https://github.com/Azure/autorest/issues/4424
`--input-file=`

**Comment by sarangan12** [comment]
Starting to work on this task. Will provide updates shortly.

**Comment by sarangan12** [comment]
@Pipeliner When you are using a OpenAPI file as input (instead of configuration file), you need to use `--input-file` option. For example, I have used the following to generate the SDK for your specified input and I am able to generate the SDK successfully.

```
C:\Users\sarajama\Projects\newproj\autorest.typescript>autorest --typescript --input-file=C:\Users\sarangan\booleanexpect.json
SourceMapConsumer.initialize is a no-op when running in node.js
AutoRest code generation utility [cli version: 3.5.1; node: v14.15.4]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
info    | AutoRest core version selected from configuration: ^3.7.0.

There is a new version of AutoRest available (3.6.1).
 > You can install the newer version with with npm install -g autorest@latest

SourceMapConsumer.initialize is a no-op when running in node.js
info    |    Loading AutoRest core      'C:\Users\sarajama\.autorest\@autorestcore@3.7.2\nodemodules\@autorest\core\dist' (3.7.2)
info    |    Installing AutoRest extension '@autorest/typescript' (latest -> 6.0.0-beta.20)
installing... [========================================] 100% | 1292/1292
info    |    Installed AutoRest extension '@autorest/typescript' (latest->6.0.0-beta.20)
info    |    Loading AutoRest extension '@autorest/modelerfour' (4.21.4->4.21.4)
warning | PreCheck/CheckDuplicateSchemas | Checking for duplicate schemas, this could take a (long) while.  Run with --verbose for more detail.
info    | Autorest.Typescript took 1801ms
info    | Generation Complete
```

**Comment by sarangan12** [comment]
No more action items pending on this task. Closing this issue.
